python/semantic_kernel/semantic_functions/chat_with_data_prompt_template.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from logging import Logger
+from typing import Optional, TypeVar
+
+from semantic_kernel.models.chat.chat_message import ChatMessage
+from semantic_kernel.semantic_functions.chat_prompt_template import ChatPromptTemplate
+from semantic_kernel.semantic_functions.prompt_template_config import (
+    PromptTemplateConfig,
+)
+from semantic_kernel.template_engine.protocols.prompt_templating_engine import (
+    PromptTemplatingEngine,
+)
+
+ChatMessageT = TypeVar("ChatMessageT", bound=ChatMessage)
+
+
+class ChatWithDataPromptTemplate(ChatPromptTemplate):
```

Comment:
This would have to be inside the connectors.ai.openai folder as it is Azure OpenAI specific